scipy/stats/tests/test_kdeoth.py

```diff
@@ -318,6 +318,51 @@ def test_kde_integer_input():
     assert_array_almost_equal(kde(x1), y_expected, decimal=6)
 
 
+@pytest.mark.parametrize("bw_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']] +
+                         ["scott", "silverman"])
+@pytest.mark.parametrize("weights_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']])
+@pytest.mark.parametrize("dataset_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']])
+@pytest.mark.parametrize("point_type", [getattr(np, dtype, None) for dtype in
+                         ['float32', 'float64', 'float96', 'float128']])
+def test_kde_output_dtype(point_type, dataset_type, weights_type, bw_type):
+    # test that for any given combination of input datatypes we get the 
+    # appropriate result datatype
+    if point_type is None:
```

Comment:
If you use the suggestion I gave for computing `_ftypes`, these checks for `None` can be removed.